src/foam/core/Action.js

```diff
@@ -134,59 +134,58 @@ foam.CLASS({
   ],
 
   methods: [
-    function andSlots(a, b) {
-      return this.ExpressionSlot.create({
-        args: [ a, b ],
-        code: function(a, b) {
-          return a && b;
-        }
-      });
-    },
 
-    function andSlotAndPromise(slot, promise) {
-      return this.andSlots(slot, this.PromiseSlot.create({
-        promise: promise
-      }));
+    function hasPermissions_(x, property) {
+      if ( ! this.permissionRequired || ! x.auth ) return Promise.resolve(true);
+      return Promise.all(this.permissionConfig[property].map(p => {
+        return x.auth.check(null, p);
+      })).then(arr => arr.every(b => b));
     },
 
-    function checkPermission(x, permission) {
-      if ( ! this.permissionConfig || ! x.auth ) return Promise.resolve(true);
-      return x.auth.check(null, permission);
-    },
-
-    function isEnabledFor(data) {
-      return this.isEnabled
-        ? foam.Function.withArgs(this.isEnabled, data)
-        : true;
-    },
-
-    function createIsEnabled$(data$) {
-      return this.createSlot_(this.isEnabled, 'enabled', data$);
+    function createPermissionlessCheckFor_(data$, property) {
+      if ( ! this[property] ) return foam.core.ConstantSlot.create({value: true});
+      return foam.core.ExpressionSlot.create({
+        obj$: data$,
```

Comment:
I don't think ExpressionSlot supports 'obj' changing, so you could just do obj: data$.get(), which would be more efficient and generate less garbage/listeners.